casper/src/test/scala/coop/rchain/casper/api/CreateBlockAPITest.scala

```diff
@@ -124,17 +142,22 @@ class CreateBlockAPITest extends FlatSpec with Matchers with EitherValues {
       .use {
         case nodes @ n1 +: n2 +: _ +: _ +: _ +: Seq() =>
           import n1.{logEff, metricEff, span, timeEff}
-          val engine = new EngineWithCasper[Task](n1.casperEff)
+          val engine                        = new EngineWithCasper[Task](n1.casperEff)
+          implicit val blockStore           = n1.blockStore
+          implicit val lastFinalizedStorage = n1.lastFinalizedStorage
+          implicit val synchronyConstraintChecker = SynchronyConstraintChecker[Effect](0)
+          implicit val lastFinalizedHeightConstraintChecker =
+            LastFinalizedHeightConstraintChecker[Effect](Long.MaxValue)
           for {
             deploys      <- (0 until 3).toList.traverse(i => basicDeployData[Task](i))
             engineCell   <- Cell.mvarCell[Task, Engine[Task]](engine)
             blockApiLock <- Semaphore[Effect](1)
 
-            b1 <- n1.publishBlock(deploys(0))(nodes: _*)
-            b2 <- n2.publishBlock(deploys(1))(nodes: _*)
-            _  <- n1.syncWith(n2)
+            b1                         <- n1.publishBlock(deploys(0))(nodes: _*)
+            b2                         <- n2.publishBlock(deploys(1))(nodes: _*)
+            _                          <- n1.syncWith(n2)
 
-            _        <- n1.casperEff.addDeploy(deploys(2))
+            _ <- n1.casperEff.addDeploy(deploys(2))
             b3Status <- createBlock(blockApiLock)(engineCell)
 
             _ = b3Status.left.value should include(
```

Comment:
@nzpr this test is now failing. Is this expected?

Reply:
Thanks for fixing this! I guess I overlooked it.

Reply:
I didn't look into test. What I meant is, with your changes should this test be adjusted or it's a newly created bug?

Reply:
test should be adjusted - `SynchronyConstraintChecker[Effect](0)` is 0 here, but it should be set to corresponding value https://github.com/rchain/rchain/blob/4997ae2fe829a7c7f199136542ea135e995be6f6/casper/src/test/scala/coop/rchain/casper/api/CreateBlockAPITest.scala#L141 in all tests. After I fixed this, now ` b3Status <- createBlock` returns `NoNewDeploys`, looking into it.

Reply:
I will add ignore for this test as now we are checking for constraints before check for deploys.